commit 4517a64d41952396dbf9f51fc73d69bd600d5a0b (HEAD -> master)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2019-05-22T18:57:15+00:00 (5 months ago)
    
    TRITON-1691 want cross-account RBAC in cloudapi
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/lib/index.js b/lib/index.js
index 4d5e1c6..9a1bd47 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -772,6 +772,18 @@ UFDS.prototype._getUser = function _getUser(filter, msg, account, cb, noCache) {
             return self.listRoles(account, filter, callback, noCache);
         };
 
+        /*
+         * The "xOnAccount" versions know how to load roles with cross-account
+         * membership. They take the "owner" account UUID as an argument (the
+         * account the role belongs to, and thus the resource being accessed).
+         */
+        result.rolesOnAccount = function rolesOnAccount(acct, callback) {
+            var filter = sprintf(
+                '(&(objectclass=sdcaccountrole)(uniquemember=%s))',
+                result.dn.toString());
+            return self.listRoles(acct, filter, callback, noCache);
+        };
+
         // Same for default roles:
         result.defaultRoles = function defaultRoles(callback) {
             var filter = sprintf(
@@ -780,6 +792,15 @@ UFDS.prototype._getUser = function _getUser(filter, msg, account, cb, noCache) {
             return self.listRoles(account, filter, callback, noCache);
         };
 
+        result.defaultRolesOnAccount =
+            function defaultRolesOnAccount(acct, callback) {
+
+            var filter = sprintf(
+                '(&(objectclass=sdcaccountrole)(uniquememberdefault=%s))',
+                result.dn.toString());
+            return self.listRoles(acct, filter, callback, noCache);
+        };
+
         vasync.parallel({
             funcs: [
                 function loadGroups(callback) {
diff --git a/package.json b/package.json
index a431941..954f0c1 100644
--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "ufds",
   "author": "Joyent (joyent.com)",
   "description": "SmartDataCenter UFDS Client API",
-  "version": "1.5.0",
+  "version": "1.6.0",
   "homepage": "https://github.com/joyent/sdc",
   "repository": {
     "type": "git",
